doc/source/train/user-guides/hyperparameter-optimization.rst

```diff
@@ -3,127 +3,168 @@
+
+* :class:`~ray.tune.Tuner` launches the tuning job, which runs a bunch of ``train_driver_fn`` trials with different hyperparameter configurations.
+* ``train_driver_fn``, which instantiates a ``TorchTrainer`` (or some other framework trainer), takes in a hyperparameter configuration and launches the distributed training job.
+* :class:`~ray.train.ScalingConfig` defines the number of training workers and resources per worker for a single Ray Train run.
+* ``train_fn_per_worker`` is the Python code that executes on each distributed training worker for a trial.
+
+.. literalinclude:: ../doc_code/train_tune_interop.py
+    :language: python
+    :start-after: __quickstart_start__
+    :end-before: __quickstart_end__
+
+
+Configuring Resources for Multiple Trials
+-----------------------------------------
+
+Ray Tune launches multiple trials which run a user-defined function in a remote Ray actor, where each trial gets a different sampled hyperparameter configuration.
+Typically, these Tune trials do work computation directly inside the Ray actor. For example, each trial could request 1 GPU and do some single-process model
+training within the remote actor itself. When using Ray Train inside Ray Tune functions, the Tune trial is actually not doing extensive computation inside this actor
+-- instead it just acts as a driver process to launch and monitor the Ray Train workers running elsewhere.
+
+.. figure:: ../images/hyperparameter_optimization/train_tune_interop.png
     :align: center
 
-    The `Tuner` will take in a `Trainer` and execute multiple training runs, each with different hyperparameter configurations.
+    Example of Ray Train runs being launched from within Ray Tune trials.
 
-Key Concepts
-------------
 
-There are a number of key concepts when doing hyperparameter optimization with a :class:`~ray.tune.Tuner`:
+.. figure:: ../images/hyperparameter_optimization/train_without_tune.png
+    :align: center
 
-* A set of hyperparameters you want to tune in a *search space*.
-* A *search algorithm* to effectively optimize your parameters and optionally use a
-  *scheduler* to stop searches early and speed up your experiments.
-* The *search space*, *search algorithm*, *scheduler*, and *Trainer* are passed to a Tuner,
-  which runs the hyperparameter tuning workload by evaluating multiple hyperparameters in parallel.
-* Each individual hyperparameter evaluation run is called a *trial*.
-* The Tuner returns its results as a :class:`~ray.tune.ResultGrid`.
+    A single Ray Train run to showcase how using Ray Tune just adds a layer of hierarchy to this tree of processes.
 
-.. note::
-   Tuners can also be used to launch hyperparameter tuning without using Ray Train. See
-   :ref:`the Ray Tune documentation <tune-main>` for more guides and examples.
 
-Basic usage
------------
+Set ``max_concurrent_trials`` to limit the number of Ray Train driver processes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
This header feels a bit technical, can simplify it to be: `Limiting the number of concurrent trials`

`max_concurrent_trials` is the answer for how to do it.